circuits/src/zk_circuits/valid_commitments.rs

```diff
@@ -198,98 +195,97 @@ where
+        let mut curr_index = cs.zero();
+        let mut balance_found = cs.false_var();
+
         for balance in wallet.balances.iter() {
-            let index_mask =
-                EqGadget::eq::<LinearCombination, Variable, _, _, _>(curr_index.clone(), index, cs);
-            let balances_eq = EqGadget::eq(balance.clone(), target_balance.clone(), cs);
+            let index_mask = EqGadget::eq(&curr_index, &index, cs)?;
+            let balances_eq = EqGadget::eq(balance, target_balance, cs)?;
+
+            let found = cs.logic_and(index_mask, balances_eq)?;
 
-            let found = AndGate::and(index_mask, balances_eq, cs);
-            balance_found += found;
-            curr_index += Variable::One();
+            balance_found = cs.logic_or(balance_found, found)?;
+            curr_index = cs.add(curr_index, cs.one())?;
         }
 
-        cs.constrain(balance_found - Variable::One())
+        cs.enforce_true(balance_found)
     }
 
     /// Verify that the wallet has the given order at the specified index
-    fn contains_order_at_index<CS: RandomizableConstraintSystem>(
+    fn contains_order_at_index(
         index: Variable,
-        target_order: OrderVar<Variable>,
-        wallet: &WalletVar<LinearCombination, MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
-        cs: &mut CS,
-    ) {
-        let mut curr_index: LinearCombination = Variable::Zero().into();
-        let mut order_found: LinearCombination = Variable::Zero().into();
+        target_order: &OrderVar,
+        wallet: &WalletVar<MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
+        cs: &mut PlonkCircuit,
+    ) -> Result<(), CircuitError> {
+        let mut curr_index = cs.zero();
+        let mut order_found = cs.false_var();
+
         for order in wallet.orders.iter() {
-            let index_mask =
-                EqGadget::eq::<LinearCombination, Variable, _, _, _>(curr_index.clone(), index, cs);
-            let orders_eq = EqGadget::eq(order.clone(), target_order.clone(), cs);
+            let index_mask = EqGadget::eq(&curr_index, &index, cs)?;
+            let orders_eq = EqGadget::eq(order, target_order, cs)?;
 
-            let found = AndGate::and(index_mask, orders_eq, cs);
-            order_found += found;
-            curr_index += Variable::One();
+            let found = cs.logic_and(index_mask, orders_eq)?;
+            order_found = cs.logic_or(order_found, found)?;
+            curr_index = cs.add(curr_index, cs.one())?;
         }
 
-        cs.constrain(order_found - Variable::One())
+        cs.enforce_true(order_found)
     }
 
     /// Verify that the wallet contains the given balance at an unspecified
     /// index
-    fn contains_balance<CS: RandomizableConstraintSystem>(
-        target_balance: BalanceVar<Variable>,
-        wallet: &WalletVar<LinearCombination, MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
-        cs: &mut CS,
-    ) {
-        let mut balance_found: LinearCombination = Variable::Zero().into();
+    fn contains_balance(
```

Comment:
nit: ik this was previously the fn name, but I think something like `constrain_balance_contained` would do better to indicate in caller fns that this adds a constraint, rather than returning a Boolean variable (also applies to other similarly named fns)